internal/controller/fluxinstance_controller.go

```diff
@@ -178,18 +204,55 @@ func (r *FluxInstanceReconciler) reconcile(ctx context.Context,
 	return requeueAfter(obj), nil
 }
 
-// build reads the distribution manifests from the storage path,
+// fetch pulls the distribution OCI artifact and
+// extracts the manifests to the temporary directory.
+// If the distribution artifact URL is not provided,
+// it falls back  to the manifests stored in the container storage.
+func (r *FluxInstanceReconciler) fetch(ctx context.Context,
+	obj *fluxcdv1.FluxInstance, tmpDir string) (string, error) {
+	log := ctrl.LoggerFrom(ctx)
+	artifactURL := obj.Spec.Distribution.Artifact
+
+	// Pull the latest manifests from the OCI repository.
+	if artifactURL != "" {
+		ctxPull, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+
+		artifactDigest, err := builder.PullArtifact(ctxPull, artifactURL, tmpDir)
```

Comment:
If I got it correctly, at every reconciliation we fetch the manifests and don't store it locally. What do you think about storing the `artifact` and its `manifest` digest, at each reconciliation, we would first fetch the `manifest` and compare the digest with local version before attempting to fetch the `blob`?

Reply:
Yes we could have a similar storage & auth implementation as SC in a followup PR. I'm also considering using the layer caching I implemented in Timoni which is more close to how OverlayFS works in OCI registries.